Boolean list is not working with PyArrow

With the following features
```
        "bool_list_feature": [
            np.array([True, False, True], dtype=bool),
            np.array([True, False, True], dtype=bool),
            np.array([True, False, True], dtype=bool),
        ],
```
The following error occurs.
```
>   ???
E   pyarrow.lib.ArrowTypeError: ('Unknown list item type: bool', 
'Conversion failed for column bool_list_feature with type object')
```

For the time being I will remove bool_list from the tests since we do support this data type through "bytes"

--- a/sdk/python/tests/dataframes.py
+++ b/sdk/python/tests/dataframes.py
@@ -116,5 +116,10 @@ ALL_TYPES = pd.DataFrame(
             np.array([b"one", b"two", b"three"]),
             np.array([b"one", b"two", b"three"]),
         ],
+        "bool_list_feature": [
+            [True, False, True],
+            [True, False, True],
+            [True, False, True],
+        ],
     }
 )

--- a/sdk/python/tests/test_client.py
+++ b/sdk/python/tests/test_client.py
@@ -986,9 +986,7 @@ class TestClient:
                 Feature(name="int32_list_feature", dtype=ValueType.INT32_LIST),
                 Feature(name="string_list_feature", dtype=ValueType.STRING_LIST),
                 Feature(name="bytes_list_feature", dtype=ValueType.BYTES_LIST),
-                # Feature(name="bool_list_feature",
-                # dtype=ValueType.BOOL_LIST), # TODO: Add support for this
-                #  type again https://github.com/feast-dev/feast/issues/341
+                Feature(name="bool_list_feature", dtype=ValueType.BOOL_LIST),
                 Feature(name="double_list_feature", dtype=ValueType.DOUBLE_LIST),
             ],
             max_age=Duration(seconds=3600),

--- a/tests/e2e/redis/all_types_parquet/all_types_parquet.yaml
+++ b/tests/e2e/redis/all_types_parquet/all_types_parquet.yaml
@@ -29,4 +29,6 @@ spec:
       valueType: STRING_LIST
     - name: bytes_list_feature_parquet
       valueType: BYTES_LIST
+    - name: bool_list_feature_parquet
+      valueType: BOOL_LIST
   maxAge: 0s

--- a/tests/e2e/redis/basic-ingest-redis-serving.py
+++ b/tests/e2e/redis/basic-ingest-redis-serving.py
@@ -886,12 +886,11 @@ def all_types_dataframe():
                 np.array([b"one", b"two", b"three"]),
                 np.array([b"one", b"two", b"three"]),
             ],
-            # "bool_list_feature": [
-            #     np.array([True, False, True]),
-            #     np.array([True, False, True]),
-            #     np.array([True, False, True]),
-            # ],
-            # TODO: https://github.com/feast-dev/feast/issues/341
+            "bool_list_feature": [
+                [True, False, True],
+                [True, False, True],
+                [True, False, True],
+            ],
         }
     )
 
@@ -918,6 +917,7 @@ def test_all_types_register_feature_set_success(client):
             Feature(name="int32_list_feature", dtype=ValueType.INT32_LIST),
             Feature(name="string_list_feature", dtype=ValueType.STRING_LIST),
             Feature(name="bytes_list_feature", dtype=ValueType.BYTES_LIST),
+            Feature(name="bool_list_feature", dtype=ValueType.BOOL_LIST),
         ],
         max_age=Duration(seconds=3600),
     )
@@ -970,6 +970,7 @@ def test_all_types_retrieve_online_success(client, all_types_dataframe):
         "string_list_feature",
         "bytes_list_feature",
         "double_list_feature",
+        "bool_list_feature",
     ]
 
     def try_get_features():
@@ -1179,11 +1180,9 @@ def all_types_parquet_file():
             "bytes_list_feature_parquet": [
                 np.array([b"one", b"two", b"three"]) for _ in range(COUNT)
             ],
+            "bool_list_feature_parquet": [[True, False, True] for _ in range(COUNT)],
         }
     )
-
-    # TODO: Boolean list is not being tested.
-    #  https://github.com/feast-dev/feast/issues/341
 
     file_path = os.path.join(tempfile.mkdtemp(), "all_types.parquet")
     df.to_parquet(file_path, allow_truncated_timestamps=True)

--- a/tests/load/ingest.py
+++ b/tests/load/ingest.py
@@ -47,6 +47,9 @@ df = pd.DataFrame(
         "bytes_list_feature": [
             np.array([b"one", b"two", b"three"]) for _ in range(number_of_entities)
         ],
+        "bool_list_feature": [
+            [True, False, True] for _ in range(number_of_entities)
+        ],
     }
 )
 
@@ -67,6 +70,7 @@ all_types_fs_expected = FeatureSet(
         Feature(name="int32_list_feature", dtype=ValueType.INT32_LIST),
         Feature(name="string_list_feature", dtype=ValueType.STRING_LIST),
         Feature(name="bytes_list_feature", dtype=ValueType.BYTES_LIST),
+        Feature(name="bool_list_feature", dtype=ValueType.BOOL_LIST),
     ],
 )
 

